docs/USAGE-GUIDE.md

```diff
@@ -1796,11 +1809,71 @@ extern int s2n_async_pkey_op_set_output(struct s2n_async_pkey_op *op, const uint
 The **op** will copy the data into a buffer passed in through the **data** parameter.
 This buffer is owned by the application, and it is the responsibility of the
 application to free it.
-**s2n_async_pkey_op_set_output** copies the inputted data buffer, and uses it
+**s2n_async_pkey_op_set_output** copies the input data buffer and uses it
 to complete the private key operation. The data buffer is owned by the application.
 Once **s2n_async_pkey_op_set_output** has returned, the application is free to
 release the data buffer.
 
+Example:
+```C
+int user_sign_or_decrypt(s2n_async_pkey_op_type op_type, uint8_t *input_buffer, size_t input_size,
+    uint8_t *output_buffer, size_t *output_size);
+
+struct user_async_pkey_op *pkey_op = NULL;
+static int user_async_pkey_cb(struct s2n_connection *conn, struct s2n_async_pkey_op *op)
+{
+    pkey_op = op;
+    return S2N_SUCCESS;
+}
+
+int negotiate(int fd, uint8_t *public_pem_bytes, size_t public_pem_size) {
+    struct s2n_cert_chain_and_key *cert_chain = s2n_cert_chain_and_key_new();
```

Comment:
So... examples are hard... On the one hand it's good to show minimally what the application needs to do to accomplish as task but I worry about this not handling any errors and getting copy-pasted into an implementation. Thoughts?

Reply:
> An unreadable example seems to defeat the purpose of having an example.

Yeah definitely agree with that. I just worry that readers will take it as an example and not handle their errors. I guess I would either suggest calling out that they wouldn't be able to use this off the shelf or remove it.